### PR TITLE
Test endpoints

### PR DESCRIPTION
**Pull request by defvol**
This PR is for adding basic endpoint tests just to guide a full coverage testing PR

My first approach is to try server injection as suggested in the Hapi-js docs: http://hapijs.com/api#serverinjectoptions-callback

This PR will be ready to merge once we have a couple of endpoint tests using different HTTP methods.

### Testing locally

Run an instance of elastic-search
```
% docker run --rm -p 9200:9200 -p 9300:9300 --name=es elasticsearch
```

Set ElasticHost env var if elastic-search host is not on localhost:9200
```
% ElasticHost=192.168.99.100:9200 npm test
```

cc: @Rub21

### CODE REVIEW

**Comment by Rub21** [comment]
@rodowi I've made all testing for the endpoints, can you take a look  them, I think after that we are ready to merge this branch!!

File: test/routesTasks.test.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    });
+  });
+});
+
+// Returns the list of existing tasks
+test('GET /tasks', function(t) {
+  setTimeout(function() {
+    server.inject('/tasks', (res) => {
+      t.equal(res.statusCode, 200, 'HTTP 200 OK');
+      t.notEqual(res.result.tasks.length, 0, 'Tasks length is greater than 0');
```
Comment (defvol) [inline test/routesTasks.test.js]: this could make undefined values pass, better check for t.true(length > 0)

File: test/routesTasks.test.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      method: 'PUT',
+      headers: form.getHeaders(),
+      payload: payload
+    }, (res) => {
+      t.equal(res.statusCode, 200, 'HTTP 200 OK');
+      t.equal(res.result.value.name, optionsUpdate.name, 'Name OK');
+      t.equal(res.result.value.description, optionsUpdate.description, 'Description OK');
+      t.equal(res.result.value.changesetComment, optionsUpdate.changesetComment, 'Changeset Comment OK');
+      t.equal(res.result.value.stats[0].items, 3, 'number of Items OK');
+      t.equal(res.result.isCompleted, false, 'Is Completed OK');
```
Comment (defvol) [inline test/routesTasks.test.js]: test message could be misleading, what bout updating it to 'task is not completed'?

File: test/routesItems.test.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      t.equal(res.statusCode, 200, 'HTTP 200 OK');
+      t.equal(res.result.type, 'Feature', 'Feature OK');
+      t.equal(res.result.properties._tofix.length, 1, 'Num Editions 1 OK');
+      t.end();
+    });
+  });
+
+  var keys;
+  var key;
+  // Group of items
```
Comment (defvol) [inline test/routesItems.test.js]: is this a GET me a record request or POST (create record)? AFAIK POST are used for creating stuff on the backend

Comment (defvol) [reply test/routesItems.test.js]: We can get into that discussion later, in the meantime expanding on the comments to make this clear would be a good idea.

File: test/routesTasks.test.js (hunk 1 above)
Comment (Rub21) [reply test/routesTasks.test.js]: this was fixed!!

File: test/routesTasks.test.js (hunk 2 above)
Comment (Rub21) [reply test/routesTasks.test.js]: fixed!!

File: test/routesItems.test.js (hunk 3 above)
Comment (Rub21) [reply test/routesItems.test.js]: @rodowi  I've added more comment on that!!

**Comment by defvol** [comment]
looks good to me
just need to figure out why travis is failing to build the tests
the error msg thrown by travis is not helpful: 'An error occurred while generating the build script.'

**Pull request merged**